client/my-sites/hosting/main.js

```diff
@@ -184,207 +183,174 @@ const SidebarCards = ( { isBasicHostingDisabled } ) => {
+	};
 
-		const getContent = () => {
-			const isGithubIntegrationEnabled =
-				isEnabled( 'github-integration-i1' ) && isAutomatticTeamMember( teams );
+	const getContent = () => {
+		const isGithubIntegrationEnabled =
+			isEnabled( 'github-integration-i1' ) && isAutomatticTeamMember( teams );
+		const WrapperComponent = ! isSiteAtomic ? FeatureExample : Fragment;
 
-			return (
-				<>
-					{ isJetpack && <QueryJetpackModules siteId={ siteId } /> }
-					{ isGithubIntegrationEnabled && (
-						<>
-							<QueryKeyringServices />
-							<QueryKeyringConnections />
-						</>
-					) }
+		return (
+			<>
+				{ isJetpack && <QueryJetpackModules siteId={ siteId } /> }
+				{ isGithubIntegrationEnabled && (
+					<>
+						<QueryKeyringServices />
+						<QueryKeyringConnections />
+					</>
+				) }
+				<WrapperComponent>
 					<Layout className="hosting__layout">
 						<Column type="main" className="hosting__main-layout-col">
 							<MainCards
 								hasStagingSitesFeature={ hasStagingSitesFeature }
-								isAdvancedHostingDisabled={ isAdvancedHostingDisabled }
-								isBasicHostingDisabled={ isBasicHostingDisabled }
+								isAdvancedHostingDisabled={ ! hasSftpFeature || ! isSiteAtomic }
+								isBasicHostingDisabled={ ! hasAtomicFeature || ! isSiteAtomic }
 								isGithubIntegrationEnabled={ isGithubIntegrationEnabled }
 								isWpcomStagingSite={ isWpcomStagingSite }
 								isMigrationTrial={ isMigrationTrial }
 								siteId={ siteId }
 							/>
 						</Column>
 						<Column type="sidebar">
-							<SidebarCards isBasicHostingDisabled={ isBasicHostingDisabled } />
+							<SidebarCards isBasicHostingDisabled={ ! hasAtomicFeature || ! isSiteAtomic } />
 						</Column>
 					</Layout>
-				</>
-			);
-		};
+				</WrapperComponent>
+			</>
+		);
+	};
 
-		/* We want to show the upsell banner for the following cases:
-		 *  1. The site does not have the Atomic feature.
-		 *  2. The site is Atomic, is not transferring, and doesn't have advanced hosting features.
-		 * Otherwise, we show the activation notice, which may be empty.
-		 */
-		const shouldShowUpgradeBanner =
-			! hasAtomicFeature ||
-			( isSiteAtomic &&
-				! isTransferInProgress() &&
-				isAdvancedHostingDisabled &&
-				! isWpcomStagingSite );
-		const banner = shouldShowUpgradeBanner ? getUpgradeBanner() : getAtomicActivationNotice();
+	/* We want to show the upsell banner for the following cases:
+	 *  1. The site does not have the Atomic feature.
+	 *  2. The site is Atomic, is not transferring, and doesn't have advanced hosting features.
+	 * Otherwise, we show the activation notice, which may be empty.
+	 */
+	const shouldShowUpgradeBanner =
+		! hasAtomicFeature || ( ! isTransferring && ! hasSftpFeature && ! isWpcomStagingSite );
+	const banner = shouldShowUpgradeBanner ? getUpgradeBanner() : getAtomicActivationNotice();
 
-		const isBusinessTrial = isMigrationTrial || isHostingTrial;
+	const isBusinessTrial = isMigrationTrial || isHostingTrial;
```

Comment:
I introduced `isSiteOnBusinessTrial` to simplfy things. I kept "Business" as it could add a lot of renaming noise and a revert seems likely based on current name-change performance.